### PR TITLE
feature#247 added abandon subsequent jobs of fail jobs

### PR DESCRIPTION
**Pull request by toan-quach**
#247

### CODE REVIEW

File: src/taipy/core/_scheduler/_scheduler.py [hunk 1]
```diff
@@ -200,11 +209,22 @@ def __remove_jobs_to_run(cls, jobs):
         cls.jobs_to_run = new_jobs_to_run
 
     @classmethod
-    def _cancel_jobs(cls, job_id_to_cancel, jobs):
+    def _fail_subsequent_jobs(cls, failed_job: Job):
+        with cls.lock:
+            to_fail_jobs = set()
```
Comment (jrobinAV) [inline src/taipy/core/_scheduler/_scheduler.py]: I would rename it to 'to_abandon_jobs'

File: src/taipy/core/_scheduler/_scheduler.py [hunk 2]
```diff
@@ -131,11 +141,10 @@ def _unlock_edit_on_outputs(jobs: Union[Job, List[Job], Set[Job]]):
...
     @classmethod
     def _on_status_change(cls, job: Job):
-        if job.is_completed():
+        if job.is_completed() or job.is_skipped():
             cls.__unblock_jobs()
-            cls.__check_and_execute_jobs_if_development_mode()
-        elif job.is_canceled():
-            cls.__check_and_execute_jobs_if_development_mode()
+        elif job.is_failed():
+            cls._fail_subsequent_jobs(job)
```
Comment (jrobinAV) [inline src/taipy/core/_scheduler/_scheduler.py:153]: When the job fails, I don't think we unlock the outputs. Do we?
Do we have a test that check that asserts that after a job fails its outputs are unlocked?

File: src/taipy/core/_scheduler/_scheduler.py (hunk 1 above)
Comment (toan-quach) [reply src/taipy/core/_scheduler/_scheduler.py]: I'll also add the failed_jobs to the set and rename it to to_fail_or_abandon_jobs :D

File: src/taipy/core/_scheduler/_scheduler.py (hunk 2 above)
Comment (toan-quach) [reply src/taipy/core/_scheduler/_scheduler.py:153]: As we discussed earlier, we're only unlocking the abandoned jobs and not the failed job. I've added some manipulation to the set to_fail_or_cancel_jobs to make sure we also unlock the failed job

**Review by jrobinAV — APPROVED** [review]

**Pull request merged**